### PR TITLE
Support JobBackoffLimitPerIndex feature gate fields

### LINKED ISSUE #2398: Support `JobBackoffLimitPerIndex` feature gate fields
### Description

Several fields should be added with `JobBackoffLimitPerIndex` graduating to Beta in v1.29

Fields

JobSpec

- [ ] backoffLimitPerIndex
- [ ] maxFailedIndexes

JobStatus
- [ ] failedIndexes

[Related PR
](https://github.com/kubernetes/kubernetes/pull/121356)

### Potential Terraform Configuration

[Reference to v1.29 Release Note
](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.29.md#v1290:~:text=Graduated%20Job%20BackoffLimitPerIndex%20feature%20to%20beta.%20(%23121356%2C%20%40mimowo))

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by theloneexplorerquest**
### Description
resolve #2398 

### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
└─(17:46:40 on Support-JobBackoffLimitPerIndex-feature-gate-fields)──> make testacc TESTARGS='-run TestAccKubernetesCronJobV1_minimalWithBackoffLimitPerIndex'
==> Checking that code complies with gofmt requirements...
go vet ./...
TF_ACC=1 go test "/Users/mau/Dev/terraform-provider-kubernetes/kubernetes" -v -vet=off -run TestAccKubernetesCronJobV1_minimalWithBackoffLimitPerIndex -parallel 8 -timeout 3h
=== RUN   TestAccKubernetesCronJobV1_minimalWithBackoffLimitPerIndex
=== PAUSE TestAccKubernetesCronJobV1_minimalWithBackoffLimitPerIndex
=== CONT  TestAccKubernetesCronJobV1_minimalWithBackoffLimitPerIndex
--- PASS: TestAccKubernetesCronJobV1_minimalWithBackoffLimitPerIndex (12.16s)
PASS
ok      github.com/hashicorp/terraform-provider-kubernetes/kubernetes   13.125s
```

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):

```release-note:enhancement
Add `backoff_per_limit_index` and `max_failed_indexes` fields in `structure_job.go`
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

**Comment by alexsomesan** [comment]
Thanks a lot for the contribution, @theloneexplorerquest

We have a long standing policy of not including features still behind feature gates, because there is no mechanism available on the Kubernetes API for the provider to query whether the feature gate is on or off and it would lead to unexpected errors on clusters that don't have the feature on.

As I mentioned, we appreciate the contribution and would love to merge this once the feature becomes universally available. Please feel free to keep the PR up to date going forward.

Thanks!

**Comment by theloneexplorerquest** [comment]
Hi @alexsomesan , 

Thanks for the update, I will keep this open and wait until this feature become universally available (and do more testing at that time!).

File: kubernetes/schema_job_spec.go [hunk 1]
```diff
@@ -57,6 +57,14 @@ func jobSpecFields(specUpdatable bool) map[string]*schema.Schema {
 			ValidateFunc: validateNonNegativeInteger,
 			Description:  "Specifies the number of retries before marking this job failed. Defaults to 6",
 		},
+		"backoff_limit_per_index": {
```
Comment (BBBmau) [inline kubernetes/schema_job_spec.go:60]: Looks like there's errors with the PR specifically due to the default value being used. Could you address these failures as well as include a test case for these new fields?

This is now supported with the feature gate being set to true by default. Let me know if there's anything unclear from adding tests.

```hcl
 === NAME  TestAccKubernetesCronJobV1_minimalWithPodFailurePolicy
    resource_kubernetes_cron_job_v1_test.go:179: Step 1/2 error: Error running apply: exit status 1
        
        Error: CronJob.batch "tf-acc-test-jk4onc8cdw" is invalid: [spec.jobTemplate.spec.backoffLimitPerIndex: Invalid value: 6: requires indexed completion mode, spec.jobTemplate.spec.maxFailedIndexes: Invalid value: 6: requires indexed completion mode]
        
          with kubernetes_cron_job_v1.test,
          on terraform_plugin_test.tf line 1, in resource "kubernetes_cron_job_v1" "test":
           1: resource "kubernetes_cron_job_v1" "test" {
```

Comment (BBBmau) [reply kubernetes/schema_job_spec.go:60]: any help needed on this @theloneexplorerquest ?

Comment (BBBmau) [reply kubernetes/schema_job_spec.go:60]: just a heads-up I'll be wrapping this up to get it merged as this is part of our `v2.33.0` milestone @theloneexplorerquest

**Review by BBBmau — APPROVED** [review]
everything is passing, they weren't prior due to needing to insure that this only runs on tests that have a cluster version of >1.29

**Pull request merged**